apps/dg/components/case_card/text_input.js

```diff
@@ -29,13 +29,16 @@ DG.React.ready(function () {
           },
 
           componentDidMount: function () {
+            window.addEventListener('mousedown', this._onMouseDown, true);
+            window.addEventListener('mouseup', this._onMouseUp, true);
```

Comment:
Out of curiosity, did you try adding these handlers directly to the React component? It seems like something like this should work:
```
input({
  ...
  onMouseDown(function(iEvent) {
    iEvent.stopPropagation();
  },
  onMouseUp(function(iEvent) {
    iEvent.stopPropagation();
  }
}) :
```

Reply:
I believe we tried this path and the React specific event handler occurred after the SproutCore handler when using this approach.  I'll double check again tomorrow.

Reply:
That's plausible. It's likely that the same may be required for touch events. For the latter case I added support for a `wantsTouch` class to `SC.RootResponder` so clients need only apply the class to let `SC.RootResponder` know to keep its mitts off. Something analogous seems like it would work here.

Reply:
@mklewandowski, I am not sure of the state of this PR. I don't understand whether the issue @kswenson mentioned is covered under this PR or not. Have we explored whether the case card has an issue with touch?

For my part, I am a bit ambivalent. The approach taken corresponds with the approach taken by the CFM, but may be incomplete. The approach @kswenson seems to be suggesting -- modifying the SproutCore RootResponder to support SC ignoring mouse events when a new dg-wants-mouse class is present in the hierarchy of current element -- is pretty clean, but involves a change to Sproutcore source. I am generally loathe to change Sproutcore if there is an alternative.

Reply:
One possible approach to migrating away from SproutCore would be to gradually replace SproutCore-based components with React-based ones like the data card. In that case, having a consistent means of indicating that the SC.RootResponder should let the component handle its own mouse events would be preferable to having each component do so independently.

Reply:
Good point. @mklewandowski , do you feel up to exploring this approach? You can find how @kswenson handled the touch events, I believe, by grepping the code for dg-wants-touch.

Reply:
@kswenson @jsandoe This approach presented by Kirk also seems to work.  I added a `dg-wants-mouse` class that will allow an element to bypass the SproutCore mouse events (mousedown, mouseup, mousemove, click, dblclick, mousewheel).  This does, however, require changes to SproutCore'S `root_responder.JS`.  It looks like the latest SproutCore that CODAP uses is located in the Concord fork on the branch dg-sc-1.11.
https://github.com/concord-consortium/codap-sproutcore/tree/dg-sc-1.11
To clarify, should I commit my SproutCore changes to the dg-sc-1.11 branch so they can be reviewed?  I want to confirm that I commit them in the expected location.